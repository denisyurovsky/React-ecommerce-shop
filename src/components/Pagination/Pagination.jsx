import Pagination from '@mui/material/Pagination';
import Stack from '@mui/material/Stack';
import PropTypes from 'prop-types';
import React from 'react';

export default function PaginationBlock({
  pageCount,
  setFilterProperties,
  className,
  currentPage,
}) {
  const handleChange = (event, value) => {
    setFilterProperties((prev) => ({
      ...prev,
      currentPage: value,
    }));
  };

  if (!pageCount) return null;

  return (
    <Stack spacing={2} className={className}>
      <Pagination
        color="primary"
        count={pageCount}
        page={currentPage}
        onChange={handleChange}
      />
    </Stack>
  );
}

PaginationBlock.propTypes = {
  setFilterProperties: PropTypes.func.isRequired,
  pageCount: PropTypes.number,
  currentPage: PropTypes.number,
  className: PropTypes.string,
};
