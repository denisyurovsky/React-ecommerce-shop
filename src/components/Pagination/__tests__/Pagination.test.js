import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import Pagination from '../Pagination';

describe('Pagination component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const setFilterProperties = jest.fn();
      const { asFragment } = render(
        <Pagination
          setFilterProperties={setFilterProperties}
          currentPage={2}
          pageCount={8}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('main group of tests', () => {
    it('Pagination buttons works properly', () => {
      let data;
      const setFilterProperties = jest.fn((func) => (data = func()));

      render(
        <Pagination
          setFilterProperties={setFilterProperties}
          currentPage={1}
          pageCount={6}
        />
      );

      const button = screen.getByRole('button', { name: /4/ });

      userEvent.click(button);
      expect(data.currentPage).toEqual(4);
    });
  });
});
