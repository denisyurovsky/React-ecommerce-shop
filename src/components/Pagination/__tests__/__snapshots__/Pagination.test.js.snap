// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1p5q5e5-MuiStack-root"
  >
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
    >
      <ul
        class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 2"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            3
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            4
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            5
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <div
            class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
          >
            â€¦
          </div>
        </li>
        <li>
          <button
            aria-label="Go to page 8"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            8
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
